devtools/maven/src/main/java/io/quarkus/maven/NativeImageMojo.java

```diff
@@ -186,278 +175,157 @@ public NativeImageMojo() {
-            appCoords = new AppArtifact(appMvnArtifact.getGroupId(), appMvnArtifact.getArtifactId(),
-                    appMvnArtifact.getClassifier(), appMvnArtifact.getExtension(),
-                    appMvnArtifact.getVersion());
-        } else {
-            final String[] coordsArr = appArtifact.split(":");
-            if (coordsArr.length < 2 || coordsArr.length > 5) {
-                throw new MojoExecutionException(
-                        "appArtifact expression " + appArtifact
-                                + " does not follow format groupId:artifactId:classifier:type:version");
-            }
-            final String groupId = coordsArr[0];
-            final String artifactId = coordsArr[1];
-            String classifier = "";
-            String type = "jar";
-            String version = null;
-            if (coordsArr.length == 3) {
-                version = coordsArr[2];
-            } else if (coordsArr.length > 3) {
-                classifier = coordsArr[2] == null ? "" : coordsArr[2];
-                type = coordsArr[3] == null ? "jar" : coordsArr[3];
-                if (coordsArr.length > 4) {
-                    version = coordsArr[4];
+        boolean clear = false;
+        try {
+
+            final Properties projectProperties = project.getProperties();
+            final Properties realProperties = new Properties();
+            for (String name : projectProperties.stringPropertyNames()) {
+                if (name.startsWith("quarkus.")) {
+                    realProperties.setProperty(name, projectProperties.getProperty(name));
                 }
             }
-            if (version == null) {
-                for (Artifact dep : project.getArtifacts()) {
-                    if (dep.getArtifactId().equals(artifactId)
-                            && dep.getGroupId().equals(groupId)
-                            && dep.getClassifier().equals(classifier)
-                            && dep.getType().equals(type)) {
-                        appMvnArtifact = new DefaultArtifact(dep.getGroupId(),
-                                dep.getArtifactId(),
-                                dep.getClassifier(),
-                                dep.getArtifactHandler().getExtension(),
-                                dep.getVersion());
-                        break;
+            realProperties.putIfAbsent("quarkus.application.name", project.getArtifactId());
+            realProperties.putIfAbsent("quarkus.application.version", project.getVersion());
+
+            Map<String, String> config = createCustomConfig();
+            Map<String, String> old = new HashMap<>();
+            for (Map.Entry<String, String> e : config.entrySet()) {
+                old.put(e.getKey(), System.getProperty(e.getKey()));
+                System.setProperty(e.getKey(), e.getValue());
+            }
+            try {
+                CuratedApplication curatedApplication = QuarkusBootstrap.builder(outputDirectory.toPath())
```

Comment:
Let's create the maven artifact resolver and pass it to the bootstrap here.

Reply:
Ok. All the resolution stuff is what I am least sure about, I think it is likely that some of my changes are not 100% correct, or at the very least not as efficient as they could be.

Reply:
First, the repo system, session, repos, etc here are properly initialized in the mojo, let's call them the original ones, because in tests and now in bootstrap we are re-creating them (currently, most probably using different versions of the api and impl and there have been lots of issues related to this initialization). Second, avoiding them being re-initialized is going to be a lot more efficient, given that Maven has already cached some artifact and resolution request-related data.